# trajectory poses
float64[] x

# trajectory derivatives
float64[] dx

# time span for dx calculation
duration dt

# time of the current x value
time t

# trajectory number
uint32	  index
